src/core/electric_utility.jl

```diff
@@ -67,27 +88,101 @@ function ElectricUtility(;
     net_metering_limit_kw::Real 
     interconnection_limit_kw::Real 
 
+
     function ElectricUtility(;
+        emissions_factor_series_lb_CO2_per_kwh::Union{Float64,Array{<:Real,1}} = Float64[],
+        emissions_factor_series_lb_NOx_per_kwh::Union{Float64,Array{<:Real,1}} = Float64[],
+        emissions_factor_series_lb_SO2_per_kwh::Union{Float64,Array{<:Real,1}} = Float64[],
+        emissions_factor_series_lb_PM25_per_kwh::Union{Float64,Array{<:Real,1}} = Float64[],
+        emissions_factor_CO2_decrease_pct::Float64 = 0.01174,
+        emissions_factor_NOx_decrease_pct::Float64 = 0.01174,
+        emissions_factor_SO2_decrease_pct::Float64 = 0.01174,
+        emissions_factor_PM25_decrease_pct::Float64 = 0.01174,
         outage_start_time_step::Int=0,  # for modeling a single outage, with critical load spliced into the baseline load ...
         outage_end_time_step::Int=0,  # ... utiltity production_factor = 0 during the outage
-        allow_simultaneous_export_import::Bool = true,  # if true the site has two meters (in effect)
-        # variables below used for minimax the expected outage cost,
+        allow_simultaneous_export_import::Bool=true,  # if true the site has two meters (in effect)
+        # next 5 variables below used for minimax the expected outage cost,
         # with max taken over outage start time, expectation taken over outage duration
         outage_start_time_steps::Array{Int,1}=Int[],  # we minimize the maximum outage cost over outage start times
         outage_durations::Array{Int,1}=Int[],  # one-to-one with outage_probabilities, outage_durations can be a random variable
-        outage_probabilities::Array{R,1} where R<:Real = [1.0],
+        outage_probabilities::Array{<:Real,1}=[1.0],
         outage_time_steps::Union{Missing, UnitRange} = isempty(outage_durations) ? missing : 1:maximum(outage_durations),
         scenarios::Union{Missing, UnitRange} = isempty(outage_durations) ? missing : 1:length(outage_durations),
         net_metering_limit_kw::Real = 0,
-        interconnection_limit_kw::Real = 1.0e9
-    )
+        interconnection_limit_kw::Real = 1.0e9,
+        latitude::Real,
+        longitude::Real,
+        time_steps_per_hour::Int = 1
+        )
+
+        region_abbr, meters_to_region = region_abbreviation(latitude, longitude)
+        emissions_region = region(region_abbr)
+        emissions_series_dict = Dict{String,Array{Float64}}()
```

Comment:
Yes, originally I had the emissions_series function returning nothing if there was a lookup error, and I had Union{Nothing,Array{<:Real,1}} as the type for emissions_factor_series_lb_<pollutant>_per_kwh in ElectricUtility to accommodate this. Then, in Scenario, I was checking to see if CO2 emissions were nothing and the user had specified a CO2 reduction % or set include_climate_in_objective to true, or if health emissions were nothing and the user had set include_health_in_objective to true. We have to error in these cases because we can't do what the user is asking for. If the user didn't specify these, then I was setting the emissions to all zeros and allowing the optimization to proceed. It looks like recently this error handling code in scenario.jl was deleted by @rathod-b (commit 6f15ad3) and Nothing was removed from the emissions factors types by you (commit f8959c1). If I revert these changes from you two and add Nothing to the type of emissions_series_dict, does that sufficiently address your concerns?

Reply:
The commit didn't address that situation, it removed my code handling it. I've added it back in, but within the ElectricUtility constructor so as to avoid Nothing in the types.